### PR TITLE
Typesript migration

### PR DESCRIPTION
**Pull request by therealedsheenan**

### CODE REVIEW

File: src/DataEditor.tsx [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
+  value: Node | string;
 };
 
 type Value = string | number;
 
 export default class DataEditor extends PureComponent<
   Types.DataEditorProps<Cell, Value>
 > {
-  input: ?HTMLInputElement;
+  input: HTMLInputElement | null | undefined;
```
Comment (dboskovic) [inline src/DataEditor.tsx:14]: just type as `input?: HTMLInputElement` - why the addition of `null`?

File: src/Spreadsheet.tsx [hunk 2]
```diff
@@ -1,22 +1,18 @@
-// @flow
-
-import React, { PureComponent } from "react";
-import type { ComponentType, Node } from "react";
+import * as React from "react";
+import { PureComponent } from "react";
```
Comment (dboskovic) [inline src/Spreadsheet.tsx:2]: Shouldn't have multiple "react" imports

File: src/Spreadsheet.tsx [hunk 3]
```diff
@@ -30,51 +26,45 @@ import * as Actions from "./actions";
...
-
-type State = {|
-  rows: number,
-  columns: number,
-  mode: Types.Mode
-|};
+export type Props<CellType, Value> = {
+  data: Matrix.Matrix<CellType>;
+  columnLabels?: string[];
+  ColumnIndicator?: React.Component<ColumnIndicatorProps>;
```
Comment (dboskovic) [inline src/Spreadsheet.tsx:38]: Suggest importing `{X}` from react rather than doing `React.X`

File: src/Spreadsheet.tsx [hunk 4]
```diff
@@ -194,25 +193,33 @@ class Spreadsheet<CellType, Value> extends PureComponent<{|
...
+        } finally {
+          done(value);
+        }
       }
-    });
+    );
     this.formulaParser.on(
       "callRangeValue",
-      (startCellCoord, endCellCoord, done) => {
+      (startCellCoord: any, endCellCoord: any, done: (_: string) => void) => {
```
Comment (dboskovic) [inline src/Spreadsheet.tsx:222]: Does this need to be `any`

File: src/SpreadsheetStateProvider.tsx [hunk 5]
```diff
@@ -109,11 +112,11 @@ export default class SpreadsheetStateProvider<
...
   }
 
-  componentDidUpdate(prevProps: Props<CellType, Value>) {
+  componentDidUpdate() {
     if (this.props.data !== this.prevState.data) {
-      this.store.setState(
-        Actions.setData(this.store.getState(), this.props.data)
-      );
+      this.store.setState({
+        data: this.props.data
```
Comment (dboskovic) [inline src/SpreadsheetStateProvider.tsx:118]: Is this a functional change?

**Review by dboskovic — CHANGES_REQUESTED** [review]

**Comment by therealedsheenan** [comment]
Removing to give way to a new clean PR.

**Pull request closed without merging**